app.rb

```diff
@@ -2,24 +2,165 @@
-student1 = Student.new('Mudasir Sherwani')
-student2 = Student.new('Ashraf')
-classroom.add_student(student1)
-classroom.add_student(student2)
-
-# Create some books and rentals
-book1 = Book.new('The Great Book', 'Fitzgerald')
-book2 = Book.new('To Kill a Tiger', 'Harper')
-book3 = Book.new('1895', 'George Orwell')
-person1 = Person.new(32, 'Alice')
-person2 = Person.new(25, 'Bob')
-Rental.new('2022-02-02', book1, person1)
-Rental.new(Date.today, book2, person1)
-Rental.new(Date.today, book3, person2)
-
-# Check rented books for a person
-puts person1.rented_books.map(&:title)
+class App
+  attr_accessor :books, :rentals, :people
+
+  def initialize
+    @books = []
+    @people = []
+    @rentals = []
+  end
+
+  def list_books
+    book_data = ''
+    @books.each do |book|
+      book.instance_variables.each do |val|
+        value = book.instance_variable_get(val)
+        val = val.to_s.delete('@')
+        val = val.capitalize
+        book_data += " #{val}: #{value}" unless val.include?('Rentals')
+      end
+      book_data += "\n"
+    end
+    if book_data == ''
+      puts 'Books Not Found !!'
+    else
+      puts book_data
+    end
+  end
+
+  def create_book
+    puts 'Title: '
+    t = gets.chomp
+    puts 'Author: '
+    a = gets.chomp
+    book = Book.new(t, a)
+    @books.push(book)
+    puts 'Book created successfully'
+  end
+
+  def list_people
+    person_data = ''
+    @people.each do |p|
+      p.instance_variables.each do |val|
+        value = p.instance_variable_get(val)
+        val = val.to_s.delete('@')
+        val = val.capitalize
+        person_data += " #{val}: #{value} " unless val.include?('Rentals') or val.include?('Classroom')
+      end
+      person_data += "\n"
+    end
+    if person_data == ''
+      puts 'No person Found !!'
+    else
+      puts person_data
+    end
+  end
+
+  def create_person
+    puts 'Choose an option student(1) or teacher(2)?'
+    choise = gets.chomp.to_i
+    return unless choise != 0
+
+    case choise
+    when 1
+      create_student
+    when 2
+      create_teacher
+    else
+      puts 'Invalid Input ( Enter 0 for Exit)'
+      create_person
+    end
+  end
+
+  def create_student
+    puts 'Age: '
+    ag = gets.chomp
+    puts 'Name: '
+    nm = gets.chomp
+    puts 'Has parent permission? [Y/N]'
+    parent_permission = gets.chomp.upcase == 'Y'
+    student = Student.new(ag, nm, parent_permission)
+    @people.push(student)
+    puts 'Student added  successfully'
+  end
+
+  def create_teacher
+    puts 'Age: '
+    ag = gets.chomp
+    puts 'Name: '
+    nm = gets.chomp
+    puts 'Specialization: '
+    specialization = gets.chomp
+    teacher = Teacher.new(ag, nm, specialization)
+    @people.push(teacher)
+    puts 'Teacher added successfully'
+  end
+
+  def create_rental
+    puts 'Select a book from the list by No'
+    books_index
+    selected_book = gets.chomp.to_i
+    puts 'Select a person from the list by No'
+    people_index
```

Comment:
- [ ] Books to select are not actually displayed in order to create a rental as you can see in the screenshot below. Let's kindly fix the issue


![Screenshot from 2023-03-10 18-27-59](https://user-images.githubusercontent.com/63087491/224383618-a8be54a2-4924-4130-9cb4-503262aca819.png)

- [ ] Then the app breaks when struggling to add a rental.



![Screenshot from 2023-03-10 18-32-58](https://user-images.githubusercontent.com/63087491/224384170-a034e808-1545-44de-9bdd-1086b6478f2c.png)